content/general/functions.textile

```diff
@@ -8,17 +8,176 @@ Reactor Functions allow your server-less functions to be invoked following chann
+
+Reactor Functions are are designed to trigger server-less functions on common platforms such as Amazon and Google.  If you are looking to trigger the execution of code on your own servers over HTTP when a message is published or event occurs, see our "Reactor WebHooks":/general/webhooks.
+
+h2. Message Format
+
+The format of messages sent can be defined in your "account dashboard":https://support.ably.io/solution/articles/3000048664-how-do-i-access-my-account-dashboard. You can have the message sent as a JSON message, or as a "MessagePack":https://msgpack.org.
+
+* JSON (JavaScript Object Notation): An efficient data-interchange format which is fairly standard and provides simple text based encoding.
+* MsgPack: An efficient binary serialization format that is similar to JSON, but faster and smaller.
+
+
+h2. Envelopes
+
+Messages sent by Reactor Functions can be contained within an envelope. This can be useful for clients which are unable to access response headers or status codes.
 
-Reactor Functions are are designed to trigger server-less functions on common platform such as Amazon and Google.  If you are looking to trigger execution of code on your own servers over HTTP when a message is published or event occurs, see our "Reactor WebHooks":/general/webhooks.
 
 h2. Payload types
 
-Ably currently supports three types of data that can be delivered via our Firehose:
+Ably currently supports two types of data that can be delivered via our Functions:
+
+* "Messages":#messages - messages trigger function calls as soon as they are published on a channel
+* "Presence events":#presence - when clients enter, update their data, or leave channels, the presence event triggers a function call
+
+You can "configure Functions":#configure from the Reactor tab in your "account dashboard":https://support.ably.io/solution/articles/3000048664-how-do-i-access-my-account-dashboard on a per app basis which can apply to one or more channels in that app. Reactor Functions can be filtered by channel naming using a regular expression.
+
+
+
+h2(#messages). Messages
+
+If you are unfamiliar with how channels work in Ably, we recommend you first review the "realtime channel documentation":/realtime/channels-messages#channels.
+
+When a message is published on a channel, a Reactor Function message event is triggered, which in turn will send a message to your server-less function.
+
+Every request will contain your message within the @body@ part of the request. The contents of the @body@ depends on the format of your messages, and whether you've used an envelope or not.
```

Comment:
Not sure what you mean by "The contents of the @body@ depends on the format of your messages, and whether you've used an envelope or not."  Can you be more specific here i.e. do you mean the encoding may vary based on the configured encoding, and the message body itself may contain the `data` payload from your messages if the rule is set up to not use an envelope, or will contain the envelope (as defined below) if configured to use enveloping